utils/githooks/pre-commit.d/20-flake.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/sh
+
+# Runs flake8 for the DAOS project.
+# Will first check uncomiited code, then either the entire tree if on master
+# or a diff against master if not.
+#
+# Picks up config files from .flake8
+
+if ! command -v flake8 > /dev/null 2>&1
+then
+    echo "No flake checking, install flake8 command"
+    exit 0
+fi
+
+echo checking uncommitted code.
+git diff -u | flake8 --diff
+
+RC=$?
+if [ $RC -ne 0 ]
+then
+    exit $RC
+fi
```

Comment:
```suggestion
if ! git diff -u | flake8 --diff; then
    exit ${PIPESTATUS[@]}
```

or even just let `set -e` do the work for you:

```suggestion
git diff -u | flake8 --diff
```

should have the same result IIRC.

Reply:
Done.